### ISSUE TITLE
Broken if package.json not in root dir

### ISSUE THREAD

**Issue by luciancic** [simonrenoult/code-complexity#14]
# Steps to reproduce
It seems like the library expects the `package.json` to be in the root dir. This is problematic if you want to use it on a larger codebase where submodules define their own `package.json`. I'm on a mac but  think the issue is platform agnostic.

```bash
mkdir test && cd test
git init
mkdir subfolder && cd subfolder
npm init -y
cd ..
npx code-complexity .
```

Output:
```
Error: Error: no package.json found, exiting.
    at Object.<anonymous> (/Users/lcondrea/.npm/_npx/81792/lib/node_modules/code-complexity/dist/src/utils.js:83:31)
    at step (/Users/lcondrea/.npm/_npx/81792/lib/node_modules/code-complexity/dist/src/utils.js:33:23)
    at Object.next (/Users/lcondrea/.npm/_npx/81792/lib/node_modules/code-complexity/dist/src/utils.js:14:53)
    at fulfilled (/Users/lcondrea/.npm/_npx/81792/lib/node_modules/code-complexity/dist/src/utils.js:5:58)
```

Try to execute from `subfolder`

```
cd subfolder
npx code-complexity .
```

Output:

```
npx: installed 64 in 4.345s
Error: Argument 'dir' must be the git root directory.
    at assertIsGitRootDirectory (/Users/lcondrea/.npm/_npx/81797/lib/node_modules/code-complexity/dist/src/lib/churn.js:111:15)
    at /Users/lcondrea/.npm/_npx/81797/lib/node_modules/code-complexity/dist/src/lib/churn.js:61:13
    at step (/Users/lcondrea/.npm/_npx/81797/lib/node_modules/code-complexity/dist/src/lib/churn.js:33:23)
    at Object.next (/Users/lcondrea/.npm/_npx/81797/lib/node_modules/code-complexity/dist/src/lib/churn.js:14:53)
    at /Users/lcondrea/.npm/_npx/81797/lib/node_modules/code-complexity/dist/src/lib/churn.js:8:71
    at new Promise (<anonymous>)
    at __awaiter (/Users/lcondrea/.npm/_npx/81797/lib/node_modules/code-complexity/dist/src/lib/churn.js:4:12)
    at compute (/Users/lcondrea/.npm/_npx/81797/lib/node_modules/code-complexity/dist/src/lib/churn.js:57:12)
    at Object.withDuration (/Users/lcondrea/.npm/_npx/81797/lib/node_modules/code-complexity/dist/src/utils.js:63:21)
    at Object.compute (/Users/lcondrea/.npm/_npx/81797/lib/node_modules/code-complexity/dist/src/lib/churn.js:53:24)
```

**Comment by simonrenoult** [comment]
Thanks for the feedback, I'll look into it!

**Comment by iamzapata** [comment]
@luciancic I had a similar issue, a monolith backend/frontend. Try something like this: 

`~/project-root/$ npx code-complexity . --filter 'client/**'`

Where the `package.json` is within `project-root` and the directory you want is `client/`

**Comment by simonrenoult** [comment]
I fixed this issue (42e835b) and released a version with the patch: https://github.com/simonrenoult/code-complexity/releases/tag/v4.0.2
